gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java

```diff
@@ -479,6 +487,20 @@ protected static String getAppEngineProjectId() {
     }
   }
 
+  protected static String serviceAccountProjectId() {
+    String project = null;
+    String credentialsPath = System.getenv("GOOGLE_APPLICATION_CREDENTIALS");
+    if(credentialsPath != null) {
+      try (InputStream credentialsStream = new FileInputStream(credentialsPath)) {
```

Comment:
That is another reason why we would like the AE to pick up first, as this would likely to fail when run on AE.